### PR TITLE
enable unit test in cadl-extension, also update surefire to M7

### PR DESCRIPTION
**Pull request by weidongxu-microsoft**

### CODE REVIEW

File: cadl-extension/pom.xml [hunk 1]
```diff
@@ -86,6 +92,16 @@
...
         <plugins>
+          <plugin>
+            <groupId>org.apache.maven.plugins</groupId>
+            <artifactId>maven-surefire-plugin</artifactId>
+            <configuration>
+              <includes>
+                <include>**/*Test.java</include>
+                <include>**/*Tests.java</include>
+              </includes>
+            </configuration>
```
Comment (weidongxu-microsoft) [inline cadl-extension/pom.xml:103]: No idea why I need to configure it again.
This should already configured in parent POM, and works fine on other modules. Maybe different groupId?
https://github.com/Azure/autorest.java/blob/bfc0dfba9256128e2ca0de4bc5aabb991b149fbd/pom.xml#L101-L105

File: pom.xml [hunk 2]
```diff
@@ -90,23 +90,25 @@
...
+              <include>**/*Test.java</include>
               <include>**/*Tests.java</include>
             </includes>
             <runOrder>alphabetical</runOrder>
             <useSystemClassLoader>false</useSystemClassLoader>
             <systemPropertyVariables combine.children="append">
-              <jacoco-agent.destfile>${project.build.directory}/jacoco.exec</jacoco-agent.destfile>
               <junit.jupiter.extensions.autodetection.enabled>true</junit.jupiter.extensions.autodetection.enabled>
+              <junit.jupiter.execution.parallel.mode.default>concurrent</junit.jupiter.execution.parallel.mode.default>
+              <junit.jupiter.execution.parallel.enabled>${parallelizeTests}</junit.jupiter.execution.parallel.enabled>
```
Comment (weidongxu-microsoft) [inline pom.xml:111]: Note this change in parent POM.

Some tests might not able to run in parallel.

File: cadl-tests/src/test/java/com/cadl/testserver/servicedriven2/DPGClientTest.java [hunk 3]
```diff
@@ -12,7 +12,7 @@
 import org.junit.jupiter.api.Test;
 
 class DPGClientTest {
-    DPGClient client = new DPGClientBuilder().buildClient();
+    ResiliencyServiceDriven2Client client = new ResiliencyServiceDriven2ClientBuilder().buildClient();
```
Comment (weidongxu-microsoft) [inline cadl-tests/src/test/java/com/cadl/testserver/servicedriven2/DPGClientTest.java:15]: @haolingdong-msft 

The updated cadl-ranch throws 404 ResourceNotFoundException in runtime, please take a look.

I might force merge this PR, as none of it changes codegen.

Comment (haolingdong-msft) [reply cadl-tests/src/test/java/com/cadl/testserver/servicedriven2/DPGClientTest.java:15]: will check

**Review by haolingdong-msft — APPROVED** [review]

**Pull request merged**